### ISSUE TITLE
Service Bus PEEK_LOCK option appears to be broken in 0.9.4

### ISSUE THREAD

**Issue by jhosteny** [Azure/azure-sdk-for-java#1075]
I updated to service bus release 0.9.4 to address [985](https://github.com/Azure/azure-sdk-for-java/pull/985). However, I now am unable to successfully receive a message with the receive option set to PEEK_LOCK. Here is a code snippet:

```java
    ReceiveMessageOptions opts = ReceiveMessageOptions.DEFAULT;
    opts.setReceiveMode(ReceiveMode.PEEK_LOCK);

    while (true) {
      try {

        BrokeredMessage brokeredMessage = this.fetchBrokeredMessage(opts);
        if (brokeredMessage == null) {
          break;
        }

        if (brokeredMessage.getMessageId() == null) {
            byte[] body = ByteStreams.toByteArray(brokeredMessage.getBody());
            String correlationId = brokeredMessage.getCorrelationId();
            LOGGER.error("BODY: " + body.toString());
            continue;
        }
```

With a newly created topic and subscription with no outstanding messages, this appears to start dequeuing a long sequence of brokered messages with no message ID, and some small sequence of bytes for the body. Here's sample output:

```
[host=MacBook-Pro-2.local ] [thread=AzureSubscriptionMessageReceiver RUNNING] timestamp=13:49:38.654 level=ERROR logger=c.b.c.i.m.a.AzureSubscriptionMessageReceiver message=TOPIC: staging-tenanteventenvelope, SUB: tenant-service-tenanteventenvelope
[host=MacBook-Pro-2.local ] [thread=AzureSubscriptionMessageReceiver RUNNING] timestamp=13:49:38.739 level=ERROR logger=c.b.c.i.m.a.BaseAzureMessageReceiver message=BODY: [B@6d455f04
[host=MacBook-Pro-2.local ] [thread=AzureSubscriptionMessageReceiver RUNNING] timestamp=13:49:38.739 level=ERROR logger=c.b.c.i.m.a.AzureSubscriptionMessageReceiver message=TOPIC: staging-tenanteventenvelope, SUB: tenant-service-tenanteventenvelope
[host=MacBook-Pro-2.local ] [thread=AzureSubscriptionMessageReceiver RUNNING] timestamp=13:49:38.829 level=ERROR logger=c.b.c.i.m.a.BaseAzureMessageReceiver message=BODY: [B@1be2aea4
[host=MacBook-Pro-2.local ] [thread=AzureSubscriptionMessageReceiver RUNNING] timestamp=13:49:38.829 level=ERROR logger=c.b.c.i.m.a.AzureSubscriptionMessageReceiver message=TOPIC: staging-tenanteventenvelope, SUB: tenant-service-tenanteventenvelope
[host=MacBook-Pro-2.local ] [thread=AzureSubscriptionMessageReceiver RUNNING] timestamp=13:49:38.929 level=ERROR logger=c.b.c.i.m.a.BaseAzureMessageReceiver message=BODY: [B@26974151
[host=MacBook-Pro-2.local ] [thread=AzureSubscriptionMessageReceiver RUNNING] timestamp=13:49:38.929 level=ERROR logger=c.b.c.i.m.a.AzureSubscriptionMessageReceiver message=TOPIC: staging-tenanteventenvelope, SUB: tenant-service-tenanteventenvelope
[host=MacBook-Pro-2.local ] [thread=AzureSubscriptionMessageReceiver RUNNING] timestamp=13:49:39.023 level=ERROR logger=c.b.c.i.m.a.BaseAzureMessageReceiver message=BODY: [B@202bce11
[host=MacBook-Pro-2.local ] [thread=AzureSubscriptionMessageReceiver RUNNING] timestamp=13:49:39.024 level=ERROR logger=c.b.c.i.m.a.AzureSubscriptionMessageReceiver message=TOPIC: staging-tenanteventenvelope, SUB: tenant-service-tenanteventenvelope
[host=MacBook-Pro-2.local ] [thread=AzureSubscriptionMessageReceiver RUNNING] timestamp=13:49:39.123 level=ERROR logger=c.b.c.i.m.a.BaseAzureMessageReceiver message=BODY: [B@20c43c66
[host=MacBook-Pro-2.local ] [thread=AzureSubscriptionMessageReceiver RUNNING] timestamp=13:49:39.123 level=ERROR logger=c.b.c.i.m.a.AzureSubscriptionMessageReceiver message=TOPIC: staging-tenanteventenvelope, SUB: tenant-service-tenanteventenvelope
[host=MacBook-Pro-2.local ] [thread=AzureSubscriptionMessageReceiver RUNNING] timestamp=13:49:39.222 level=ERROR logger=c.b.c.i.m.a.BaseAzureMessageReceiver message=BODY: [B@5cb5ce8f
[host=MacBook-Pro-2.local ] [thread=AzureSubscriptionMessageReceiver RUNNING] timestamp=13:49:39.222 level=ERROR logger=c.b.c.i.m.a.AzureSubscriptionMessageReceiver message=TOPIC: staging-tenanteventenvelope, SUB: tenant-service-tenanteventenvelope
[host=MacBook-Pro-2.local ] [thread=AzureSubscriptionMessageReceiver RUNNING] timestamp=13:49:40.800 level=ERROR logger=c.b.c.i.m.a.BaseAzureMessageReceiver message=BODY: [B@558d4dcf
[host=MacBook-Pro-2.local ] [thread=AzureSubscriptionMessageReceiver RUNNING] timestamp=13:49:40.800 level=ERROR logger=c.b.c.i.m.a.AzureSubscriptionMessageReceiver message=TOPIC: staging-tenanteventenvelope, SUB: tenant-service-tenanteventenvelope
```

If a message is queued to the topic, it never is dequeued by this code. I never see the debug logging that shows the message, and it appears on the subscription as seen from the portal.

If I roll back the service bus dependency to 0.9.3, I am again able to successfully use PEEK_LOCK, but this causes other problems since jersey 1.13 has compatibility difficulties with Dropwizard.

This seems behavior consistent with another, older report in [644](https://github.com/Azure/azure-sdk-for-java/issues/644) that reports needing to roll back the jersey dependency in order to get PEEK_LOCK to work.

**Comment by bearrito** [comment]
Affected by this as well in similar environment.

**Comment by yvgopal** [comment]
Can you guys update to v0.9.5? The issue is fixed in the release.

**Comment by jianghaolu** [comment]
Fixed in 0.9.5.
